pkg/protocol/engine/filter/blockfilter/filter.go

```diff
@@ -68,102 +67,95 @@ func New(apiProvider api.Provider, opts ...options.Option[Filter]) *Filter {
-	if timeDelta < -f.optsMaxAllowedWallClockDrift {
-		f.events.BlockFiltered.Trigger(&filter.BlockFilteredEvent{
+	minCommittableAge := blockSlotAPI.ProtocolParameters().MinCommittableAge()
+	maxCommittableAge := blockSlotAPI.ProtocolParameters().MaxCommittableAge()
+
+	// check that commitment is within allowed range.
+	if minCommittableAge > 0 &&
+		block.ProtocolBlock().SlotCommitmentID.Index() > 0 &&
+		(block.ProtocolBlock().SlotCommitmentID.Index() > block.ID().Index() ||
+			block.ID().Index()-block.ProtocolBlock().SlotCommitmentID.Index() < minCommittableAge) {
+		f.events.BlockPreFiltered.Trigger(&filter.BlockPreFilteredEvent{
 			Block:  block,
-			Reason: ierrors.Wrapf(ErrBlockTimeTooFarAheadInFuture, "issuing time ahead %s vs %s allowed", -timeDelta, f.optsMaxAllowedWallClockDrift),
+			Reason: ierrors.Wrapf(ErrCommitmentTooRecent, "block at slot %d committing to slot %d", block.ID().Index(), block.ProtocolBlock().SlotCommitmentID.Index()),
 			Source: source,
 		})
 
 		return
 	}
+	if block.ID().Index() >= block.ProtocolBlock().SlotCommitmentID.Index() && block.ID().Index()-block.ProtocolBlock().SlotCommitmentID.Index() > maxCommittableAge {
+		f.events.BlockPreFiltered.Trigger(&filter.BlockPreFilteredEvent{
+			Block:  block,
+			Reason: ierrors.Wrapf(ErrCommitmentTooOld, "block at slot %d committing to slot %d, max committable age %d", block.ID().Index(), block.ProtocolBlock().SlotCommitmentID.Index(), maxCommittableAge),
+			Source: source,
+		})
 
-	if basicBlock, isBasic := block.BasicBlock(); isBasic {
+		return
+	}
 
-		// Verify that if CommitmentInput referenced by a transaction is not too old.
-		// This check provides that Time-lock and other time-based checks in the VM don't have to rely on what is the relation of transaction to
-		// current wall time, because it will only receive transactions that are created in a recent and limited past.
-		if basicBlock.Payload != nil && basicBlock.Payload.PayloadType() == iotago.PayloadTransaction {
-			transaction, _ := basicBlock.Payload.(*iotago.Transaction)
-			if commitmentInput := transaction.CommitmentInput(); commitmentInput != nil {
-				// commitmentInput must reference a commitment
-				// that is between the latest possible, non-evicted committable slot in relation to the block time
-				// block.ID().Index() - (2 * protocolParams.EvictionAge)
-				// and the slot that block is committing to.
-
-				// Parameters moved to the other side of inequality to avoid underflow errors with subtraction from an uint64 type.
-				if commitmentInput.CommitmentID.Index()+(protocolParams.EvictionAge()<<1) < block.ID().Index() {
-					f.events.BlockFiltered.Trigger(&filter.BlockFilteredEvent{
+	// check that commitment input (if any) is within allowed range.
+	if basicBlock, isBasic := block.BasicBlock(); isBasic {
+		if tx, isTX := basicBlock.Payload.(*iotago.Transaction); isTX {
+			if cInput := tx.CommitmentInput(); cInput != nil {
+				if minCommittableAge > 0 &&
```

Comment:
So if `minCommittableAge` is zero, or an input commitment index is zero, blocks never get filtered? Is that ok?
Maybe also add a variable here for `commitmentIndex` and add some comments.

Reply:
It never get's filtered for being too recent if minCA or commitment index are zero, but it may get filtered for being too old as below with maxCA check.